<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>965</width>
    <height>652</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Image Viewer</string>
  </property>
  <property name="windowIcon">
   <iconset resource="sources.qrc">
    <normaloff>:/icons/icons/ImageViewer.png</normaloff>:/icons/icons/ImageViewer.png</iconset>
  </property>
  <property name="statusTip">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QWidget" name="rotationWidget" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="rotateLabel">
           <property name="text">
            <string>rotation angle:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="angleSpinBox">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>10</height>
            </size>
           </property>
           <property name="baseSize">
            <size>
             <width>100</width>
             <height>0</height>
            </size>
           </property>
           <property name="frame">
            <bool>false</bool>
           </property>
           <property name="buttonSymbols">
            <enum>QAbstractSpinBox::UpDownArrows</enum>
           </property>
           <property name="specialValueText">
            <string notr="true"/>
           </property>
           <property name="suffix">
            <string notr="true"/>
           </property>
           <property name="prefix">
            <string notr="true"/>
           </property>
           <property name="minimum">
            <number>-180</number>
           </property>
           <property name="maximum">
            <number>180</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="angleHSlider">
           <property name="minimum">
            <number>-180</number>
           </property>
           <property name="maximum">
            <number>180</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="zoomWidget" native="true">
        <layout class="QHBoxLayout" name="zoomHLayout">
         <item>
          <widget class="QLabel" name="zoomLabel">
           <property name="text">
            <string>zoom percentage:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="zoomSpinBox">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>10</height>
            </size>
           </property>
           <property name="baseSize">
            <size>
             <width>100</width>
             <height>0</height>
            </size>
           </property>
           <property name="frame">
            <bool>false</bool>
           </property>
           <property name="buttonSymbols">
            <enum>QAbstractSpinBox::UpDownArrows</enum>
           </property>
           <property name="specialValueText">
            <string notr="true"/>
           </property>
           <property name="suffix">
            <string notr="true"/>
           </property>
           <property name="prefix">
            <string notr="true"/>
           </property>
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>2000000</number>
           </property>
           <property name="singleStep">
            <number>10</number>
           </property>
           <property name="value">
            <number>100</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="zoomHSlider">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>900</number>
           </property>
           <property name="value">
            <number>100</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGraphicsView" name="graphicsView">
        <property name="cursor" stdset="0">
         <cursorShape>OpenHandCursor</cursorShape>
        </property>
        <property name="mouseTracking">
         <bool>false</bool>
        </property>
        <property name="autoFillBackground">
         <bool>false</bool>
        </property>
        <property name="frameShape">
         <enum>QFrame::NoFrame</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
        <property name="lineWidth">
         <number>0</number>
        </property>
        <property name="backgroundBrush">
         <brush brushstyle="NoBrush">
          <color alpha="0">
           <red>0</red>
           <green>0</green>
           <blue>0</blue>
          </color>
         </brush>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>100</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="allowedAreas">
    <set>Qt::NoToolBarArea</set>
   </property>
   <property name="orientation">
    <enum>Qt::Horizontal</enum>
   </property>
   <property name="iconSize">
    <size>
     <width>40</width>
     <height>30</height>
    </size>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonFollowStyle</enum>
   </property>
   <property name="floatable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
   <addaction name="actionZoomInArea"/>
   <addaction name="actionZoomFactor"/>
   <addaction name="actionZoomIn"/>
   <addaction name="actionZoomOut"/>
   <addaction name="actionHandTool"/>
   <addaction name="separator"/>
   <addaction name="actionRotate"/>
   <addaction name="actionCrop"/>
   <addaction name="separator"/>
   <addaction name="actionUndo"/>
   <addaction name="actionRedo"/>
   <addaction name="actionReset"/>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>965</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>file</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>view</string>
    </property>
    <addaction name="actionZoomInArea"/>
    <addaction name="actionZoomFactor"/>
    <addaction name="actionZoomIn"/>
    <addaction name="actionZoomOut"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>edit</string>
    </property>
    <addaction name="actionRotate"/>
    <addaction name="actionCrop"/>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
    <addaction name="actionReset"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
   <addaction name="menuEdit"/>
  </widget>
  <action name="actionOpen">
   <property name="icon">
    <iconset>
     <normalon>:/icons/icons/open.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="toolTip">
    <string>Open new image file (Ctrl+O)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset>
     <normalon>:/icons/icons/save.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Save as new file (Ctrl+S)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionZoomInArea">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="sources.qrc">
     <normaloff>:/icons/icons/zoomin_area.png</normaloff>:/icons/icons/zoomin_area.png</iconset>
   </property>
   <property name="text">
    <string>zoom in area</string>
   </property>
   <property name="toolTip">
    <string>select area to zoom in (Z)</string>
   </property>
   <property name="shortcut">
    <string>Z</string>
   </property>
  </action>
  <action name="actionZoomOut">
   <property name="icon">
    <iconset>
     <normalon>:/icons/icons/zoomout.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>ZoomOut</string>
   </property>
   <property name="toolTip">
    <string>zoom out (-)</string>
   </property>
   <property name="shortcut">
    <string>-</string>
   </property>
  </action>
  <action name="actionCrop">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset>
     <normalon>:/icons/icons/crop.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Crop</string>
   </property>
   <property name="toolTip">
    <string>crop to selected area (Ctrl+C)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionReset">
   <property name="text">
    <string>Reset</string>
   </property>
   <property name="toolTip">
    <string>restore orignal image (Ctrl+R)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionHandTool">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="sources.qrc">
     <normaloff>:/icons/icons/handtool.png</normaloff>:/icons/icons/handtool.png</iconset>
   </property>
   <property name="text">
    <string>hand tool</string>
   </property>
   <property name="toolTip">
    <string>hand tool (Esc, H)</string>
   </property>
   <property name="shortcut">
    <string>H</string>
   </property>
  </action>
  <action name="actionZoomIn">
   <property name="icon">
    <iconset resource="sources.qrc">
     <normaloff>:/icons/icons/zoomin.png</normaloff>:/icons/icons/zoomin.png</iconset>
   </property>
   <property name="text">
    <string>ZoomIn</string>
   </property>
   <property name="toolTip">
    <string>Zoom in (+)</string>
   </property>
   <property name="shortcut">
    <string>=</string>
   </property>
  </action>
  <action name="actionRotate">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="sources.qrc">
     <normaloff>:/icons/icons/rotate.png</normaloff>:/icons/icons/rotate.png</iconset>
   </property>
   <property name="text">
    <string>rotate</string>
   </property>
   <property name="toolTip">
    <string>rotate image</string>
   </property>
  </action>
  <action name="actionZoomFactor">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="sources.qrc">
     <normaloff>:/icons/icons/zoom_range.png</normaloff>:/icons/icons/zoom_range.png</iconset>
   </property>
   <property name="text">
    <string>zoom range</string>
   </property>
   <property name="toolTip">
    <string>zoom by certain value</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="icon">
    <iconset resource="sources.qrc">
     <normaloff>:/icons/icons/redo.png</normaloff>:/icons/icons/redo.png</iconset>
   </property>
   <property name="text">
    <string>redo</string>
   </property>
   <property name="toolTip">
    <string>redo (Shift + Ctrl + Z)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Z</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="icon">
    <iconset resource="sources.qrc">
     <normaloff>:/icons/icons/undo.png</normaloff>:/icons/icons/undo.png</iconset>
   </property>
   <property name="text">
    <string>undo</string>
   </property>
   <property name="toolTip">
    <string>undo (Ctrl + Z)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="sources.qrc"/>
 </resources>
 <connections/>
</ui>
